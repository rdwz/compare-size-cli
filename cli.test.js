import { execa } from 'execa'
import { describe, it, expect } from 'vitest'
import packageJson from './package.json'

describe('cli', () => {
	it('displays the correct help message', async () => {
		const { stdout } = await execa('./cli.js', ['--help'])
		expect(stdout).toMatchSnapshot('help-message')
	})

	it('displays the correct version', async () => {
		const { stdout } = await execa('./cli.js', ['--version'])
		expect(stdout).toBe(packageJson.version)
	})
})
