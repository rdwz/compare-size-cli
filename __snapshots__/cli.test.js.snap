// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cli > displays the correct help message > help-message 1`] = `
"
  Compare the size of two files easily ðŸ“Š

  Usage
    $ compare-size <file> <file>

  Example
    $ compare-size foo.zip bar.tar.gz
    foo.zip: 1.34 kB
    bar.tar.gz: 1 kB
    Difference: 0.34 kB
"
`;
